src/navigation/navigationhandler.cpp

```diff
@@ -328,23 +326,11 @@ void NavigationHandler::updateCameraTransitions() {
     }
 }
 
-
-void NavigationHandler::setEnableKeyFrameInteraction() {
-    _useKeyFrameInteraction = true;
-}
-
-void NavigationHandler::setDisableKeyFrameInteraction() {
-    _useKeyFrameInteraction = false;
-}
-
-void NavigationHandler::triggerPlaybackStart() {
-    _playbackModeEnabled = true;
-}
-
-void NavigationHandler::stopPlayback() {
+void NavigationHandler::handleSessionRecordingPlaybackEnd() {
+    // When the OrbitalNavigator takes over after ha session recording has finished
+    // playing, we need to make sure that it's update variables are updated correctly
     _orbitalNavigator.resetVelocities();
-    _orbitalNavigator.resetNodeMovements();
-    _playbackModeEnabled = false;
+    _orbitalNavigator.updatePreviousStateVariables();
```

Comment:
Just a suggestion: the 2 lines in this function are grouped together in `applyNavigationState` also. You could rename this function and call it from there. Renaming it would make sense because even though it's called from session recording, it isn't specific to that.

Reply:
I'm aware, and I agree. Do you maybe have a good suggestion for a name ? 😄 Because my only idea so far is something like `resetVelocitiesAndUpdateStateVariables` and that just seemed silly.

Reply:
I don't have an idea, but I don't mind longer, descriptive names, so your suggestion doesn't sound as silly to me :)

Reply:
Haha, I see. I'll give it another thought and otherwise go with it :) It would make it clearar that this is not something we only want to do when the session recording ends but in other cases too